applications/DamApplication/custom_constitutive/thermal_linear_elastic_3D_law.cpp

```diff
@@ -107,13 +97,35 @@ void ThermalLinearElastic3DLaw::CalculateMaterialResponseKirchhoff (Parameters&
         this->CalculateThermalStrain(StrainVector,ElasticVariables,Temperature);
 
         StressVector = prod(ConstitutiveMatrix,StrainVector);
-    }
-    else if(Options.Is( ConstitutiveLaw::COMPUTE_STRAIN ) ) //This should be COMPUTE_THERMAL_STRAIN
-    {
+      }
+      else{
+
+	this->CalculateLinearElasticMatrix( ConstitutiveMatrix, YoungModulus, PoissonCoefficient );
+
         double Temperature;
         this->CalculateDomainTemperature( ElasticVariables, Temperature);
+
+        Vector ThermalStrainVector;
+        this->CalculateThermalStrain(ThermalStrainVector,ElasticVariables,Temperature);
+
+        StressVector = prod(ConstitutiveMatrix,(StrainVector - ThermalStrainVector));
```

Comment:
In the end I've done the following:
`
Vector tmp(StrainVector.size());
noalias(tmp) = StrainVector - ThermalStrainVector;
noalias(StressVector) = prod(ConstitutiveMatrix,tmp);
`